libs/libc/stdio/lib_getchar.c

```diff
@@ -38,3 +38,13 @@ int getchar(void)
   return read(STDIN_FILENO, &c, 1) == 1 ? c : EOF;
 #endif
 }
+
+int getchar_unlocked(void)
+{
+#ifdef CONFIG_FILE_STREAM
+  return fgetc_unlocked(stdin);
+#else
+  unsigned char c;
+  return read(STDIN_FILENO, &c, 1) == 1 ? c : EOF;
```

Comment:
Let's change to
```suggestion
  return _NX_READ(STDIN_FILENO, &c, 1) == 1 ? c : EOF;
```
here and in `getchar`

Reply:
kernel don't call getchar and putchar, only userspace call them, so keey read? ditto.